src/partial-result-stream.ts

```diff
@@ -355,13 +359,25 @@ export function partialResultStream(
     },
   });
 
+  // This listener ensures that the last request that executed successfully
+  // after one or more retries will end the requestsStream.
+  const endListener = () => {
+    if (lastErr) {
```

Comment:
Currently, `lastErr` includes retryable and non-retryable errors. Shouldn't `lastErr` just be retryable errors?

Reply:
Yes, that's true. We don't need it for non-retryable errors. Fixed.